cdap-security/src/test/java/co/cask/cdap/security/impersonation/UGIProviderTest.java

```diff
@@ -140,97 +155,205 @@ public static void finish() {
+                        eveUGIWithPrincipal.getUGI().getAuthenticationMethod());
+    Assert.assertTrue(eveUGIWithPrincipal.getUGI().hasKerberosCredentials());
+    Assert.assertEquals(eveKerberosPrincipalId.getPrincipal(), eveUGIWithPrincipal.getPrincipal());
+
+    // cleanup
+    ownerAdmin.delete(aliceEntity);
+    ownerAdmin.delete(bobEntity);
+    namespaceClient.delete(new NamespaceId("ugiProviderNs"));
   }
 
-//  @Test
-//  public void testRemoteUGIProvider() throws Exception {
-//    // Starts a mock server to handle remote UGI requests
-//    final NettyHttpService httpService = NettyHttpService.builder("remoteUGITest")
-//      .addHttpHandlers(Collections.singleton(new UGIProviderTestHandler()))
-//      .build();
-//
-//    httpService.startAndWait();
-//    try {
-//      InMemoryDiscoveryService discoveryService = new InMemoryDiscoveryService();
-//      discoveryService.register(new Discoverable(Constants.Service.APP_FABRIC_HTTP, httpService.getBindAddress()));
-//
-//      // Create Alice UGI
-//      RemoteUGIProvider ugiProvider = new RemoteUGIProvider(cConf, discoveryService, locationFactory);
-//      ImpersonationInfo aliceInfo = new ImpersonationInfo(getPrincipal("alice"), keytabFile.toURI().toString());
-//      UserGroupInformation aliceUGI = ugiProvider.getConfiguredUGI(aliceInfo);
-//
-//      // Shouldn't be a kerberos UGI
-//      Assert.assertFalse(aliceUGI.hasKerberosCredentials());
-//      // Validate the credentials
-//      Token<? extends TokenIdentifier> token = aliceUGI.getCredentials().getToken(new Text("principal"));
-//      Assert.assertArrayEquals(aliceInfo.getPrincipal().getBytes(StandardCharsets.UTF_8), token.getIdentifier());
-//      Assert.assertArrayEquals(aliceInfo.getPrincipal().getBytes(StandardCharsets.UTF_8), token.getPassword());
-//      Assert.assertEquals(new Text("principal"), token.getKind());
-//      Assert.assertEquals(new Text("service"), token.getService());
-//
-//      token = aliceUGI.getCredentials().getToken(new Text("keytab"));
-//      Assert.assertArrayEquals(aliceInfo.getKeytabURI().getBytes(StandardCharsets.UTF_8), token.getIdentifier());
-//      Assert.assertArrayEquals(aliceInfo.getKeytabURI().getBytes(StandardCharsets.UTF_8), token.getPassword());
-//      Assert.assertEquals(new Text("keytab"), token.getKind());
-//      Assert.assertEquals(new Text("service"), token.getService());
-//
-//      // Fetch it again, it should return the same UGI due to caching
-//      Assert.assertSame(aliceUGI, ugiProvider.getConfiguredUGI(aliceInfo));
-//
-//      // Invalid the cache and fetch it again. A different UGI should be returned
-//      ugiProvider.invalidCache();
-//      Assert.assertNotSame(aliceUGI, ugiProvider.getConfiguredUGI(aliceInfo));
-//
-//    } finally {
-//      httpService.stopAndWait();
-//    }
-//  }
+  @Test
+  public void testDefaultUGIProviderWithHDFSFiles() throws Exception {
+    Location hdfsKeytabDir = locationFactory.create("keytabs");
```

Comment:
looks like there is a lot of duplication between this and the previous test case... Can you consolidate them? Fine to do it in a new PR if you want to get this merged.